lib/CMakeLists.txt

```diff
@@ -6,7 +6,11 @@ if (HCC_RUNTIME_DEBUG)
   add_definitions(-DKALMAR_DEBUG=1)
 endif (HCC_RUNTIME_DEBUG)
 
-include_directories(${CMAKE_CURRENT_BINARY_DIR} ${LIBCXX_HEADER} )
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+
+if (USE_LIBCXX)
+  include_directories(${LIBCXX_HEADER})
```

Comment:
You should use `target_include_directories` for each target that needs libc++. This way when I say `target_link_libraries(myLib hcc::mcwamp_hsa hcc::hc_am)` it will add the includes for libc++ if necessary.